From 78ca9a6aa6759597cad7723c2cd9b80059c8af3c Mon Sep 17 00:00:00 2001
From: Cody Peter Mello <cody.mello@joyent.com>
Date: Mon, 21 Nov 2016 22:43:02 +0000
Subject: [PATCH] MORAY-378 Bucket cache isn't cleared on bucket update or
 delete Reviewed by: Patrick Mooney <patrick.mooney@joyent.com> Approved by:
 Patrick Mooney <patrick.mooney@joyent.com>

---
 lib/buckets/common.js | 9 +++++++++
 lib/buckets/del.js    | 3 ++-
 lib/buckets/update.js | 4 ++--
 3 files changed, 13 insertions(+), 3 deletions(-)

diff --git a/lib/buckets/common.js b/lib/buckets/common.js
index 1990dbb..fdc8d75 100644
--- a/lib/buckets/common.js
+++ b/lib/buckets/common.js
@@ -14,6 +14,8 @@ var assert = require('assert-plus');
 var once = require('once');
 var vasync = require('vasync');
 
+var objCommon = require('../objects/common');
+
 var mod_errors = require('../errors');
 var InvalidBucketConfigError = mod_errors.InvalidBucketConfigError;
 var InvalidBucketNameError = mod_errors.InvalidBucketNameError;
@@ -243,6 +245,12 @@ function validateBucket(req, cb) {
 }
 
 
+function shootdownBucket(req, cb) {
+    objCommon.shootdownBucket(req);
+    cb();
+}
+
+
 
 ///--- Exports
 
@@ -253,5 +261,6 @@ module.exports = {
     buildIndexString: buildIndexString,
     createIndexes: createIndexes,
     mapIndexType: mapIndexType,
+    shootdownBucket: shootdownBucket,
     validateBucket: validateBucket
 };
diff --git a/lib/buckets/del.js b/lib/buckets/del.js
index 85fc8ad..59375e2 100644
--- a/lib/buckets/del.js
+++ b/lib/buckets/del.js
@@ -193,7 +193,8 @@ function del(options) {
                 deleteConfig,
                 dropTable,
                 dropSequence,
-                dropLockingSerial
+                dropLockingSerial,
+                common.shootdownBucket
             ]
         });
     }
diff --git a/lib/buckets/update.js b/lib/buckets/update.js
index b7ad783..25c5985 100644
--- a/lib/buckets/update.js
+++ b/lib/buckets/update.js
@@ -17,7 +17,6 @@ var clone = require('clone');
 
 var common = require('./common');
 var control = require('../control');
-var objCommon = require('../objects/common');
 
 var mod_errors = require('../errors');
 var BucketNotFoundError = mod_errors.BucketNotFoundError;
@@ -484,7 +483,8 @@ function update(options) {
                 dropColumns,
                 addColumns,
                 createIndexes,
-                createUniqueIndexes
+                createUniqueIndexes,
+                common.shootdownBucket
             ]
         });
     }
-- 
2.21.0

